### PR TITLE
Instrumenting available capacity

### PR DESCRIPTION
**Pull request by dannykopping**
In our setup, we have multiple servers with drone runners servicing CI jobs. At present, we cannot know how busy or quiet a particular drone runner is.

This PR introduces Prometheus metrics exposition to determine what the available and configured capacity is.
We've been running this in production for the last couple weeks and it appears to be accurate.

### CODE REVIEW

File: command/daemon/daemon.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package daemon
 
 import (
 	"context"
+	"github.com/drone/drone-go/drone"
```
Comment (marko-gacesa) [inline command/daemon/daemon.go:9]: This line should go somewhere between lines 18 and 30. Let the goimports sort them out.

File: command/daemon/metrics.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package daemon
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+	"net/http"
```
Comment (marko-gacesa) [inline command/daemon/metrics.go:8]: Move this import as the first one. And separate it from the rest with a blank line.

**Comment by marko-gacesa** [comment]
Thank you @dannykopping for your effort. We are not going to merge this PR.

Adding this only to docker runner would be a step in wrong direction because docker runner would diverge from other runners. A better solution would be to move the code to runner-go repo, so that all runners can benefit from this.

But it would be even better to have runners report their usage data back to drone server. That way it would be possible to have a nice page on UI what shows available runners and their usage statistics (in addition to metrics data for Prometheus).

We have a repo drone/proposal that serves as a place where we can discuss design proposals. I suggest you check it out. Rather than rushing to create quick solutions here, a PR that describes a problem should be created at the proposal repo, so that we all can discuss it and make it fit everyone. Thanks.

https://discourse.drone.io/t/drone-proposals/9415

**Comment by julienduchesne** [comment]
I started a proposal for this: https://github.com/drone/proposal/pull/6

**Pull request closed without merging**